CHANGELOG.md

```diff
@@ -4,6 +4,179 @@ All notable changes to Router will be documented in this file.
++ You can now override the status of a span using `otel.status_code` attribute to change the status of a span
+
+For example, by default spans are marked in error if you have a critical error or http status code != 200, now if you want to mark your span in error if you have a graphql error in response body for example then you can have this kind of configuration:
+
+```yaml
+telemetry:
+  instrumentation:
+    spans:
+      router:
+        attributes:
+          otel.status_code:
+            static: error
+            condition:
+              eq:
+              - true
+              - on_graphql_error: true
+      supergraph:
+        attributes:
+          otel.status_code:
+            static: error
+            condition:
+              eq:
+              - true
+              - on_graphql_error: true
+``` 
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/5221
+
+### Instrument incremented on aborted request when condition is not fulfilled ([PR #5215](https://github.com/apollographql/router/pull/5215))
+
+Previously when a telemetry instrument was dropped it would be incremented even if the associated condition was not fulfilled. For instance:
+
+```yaml
+telemetry:
+  instrumentation:
+    instruments:
+      router:
+        http.server.active_requests: false
+        http.server.request.duration: false
+        "custom_counter":
+          description: "count of requests"
+          type: counter
+          unit: "unit"
+          value: unit
+          # This instrument should not be triggered as the condition is never true
+          condition:
+            eq:
+              - response_header: "never-received"
+              - static: "true"
+```
+
+In the case where a request was started, but the client aborted the request before the response was sent, the `response_header` would never be set to `"never-received"`,
+and the instrument would not be triggered. However, the instrument would still be incremented.
+
+Conditions are now checked for aborted requests, and the instrument is only incremented if the condition is fulfilled.
+
+By [@BrynCooke](https://github.com/BrynCooke) in https://github.com/apollographql/router/pull/5215
+
+## 🛠 Maintenance
+
+### Send query planner and lifecycle metrics to Apollo ([PR #5267](https://github.com/apollographql/router/pull/5267), [PR #5270](https://github.com/apollographql/router/pull/5270))
+
+To allow us to measure how much of an improvement the new query planner implementation makes, we are now transmitting to Apollo new metrics:
+- `apollo.router.query_planning.*`  will greatly help us to improve query planning in the Router.
+- `apollo.router.lifecycle.api_schema` provides us feedback on the experimental Rust-based API schema generation.
+- `apollo.router.lifecycle.license` provides metrics on license expiration. We use this to improve the reliability of the license check mechanism.
+
+These metrics do not leak any sensitive information.
```

Comment:
```suggestion
To enable the performance measurement of the router's new query planner implementation, the router transmits to Apollo the following new metrics:
- `apollo.router.query_planning.*` provides metrics on the query planner that help improve the query planning implementation.
- `apollo.router.lifecycle.api_schema` provides feedback on the experimental Rust-based API schema generation.
- `apollo.router.lifecycle.license` provides metrics on license expiration that help improve the reliability of the license check mechanism.

These metrics don't leak any sensitive data.
```

Reply:
Re: `sensitive data`, is the type data sent by these metrics already of these  it covered in our [GraphOS data privacy docs](https://www.apollographql.com/docs/graphos/data-privacy)? If yes, can link it here, and if not, can add more context here.

Reply:
I believe the thing @Geal is referring to is covered by the "metrics" that we already call out on this page, so we should be fine — https://www.apollographql.com/docs/graphos/data-privacy/#current-urls